src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2163,6 +2163,32 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                 return;
             }
 
+            if (reason == LambdaConversionResult.MismatchedParameterDefaultValue)
+            {
+                Debug.Assert(anonymousFunction.ParameterCount == delegateParameters.Length);
+
+                var lambdaSymbol = anonymousFunction.TemporaryLambdaSymbol;
+                Debug.Assert(lambdaSymbol is not null);
+
+                // The lambda symbol may have diagnostics from
+                // binding default parameters, so copy these to the current diagnostic bag
+                lambdaSymbol.GetDeclarationDiagnostics(diagnostics);
+
+                for (int i = 0; i < anonymousFunction.ParameterCount; i++)
+                {
+                    var lambdaParameterLocation = anonymousFunction.ParameterLocation(i);
+                    var lambdaParamDefaultVal = lambdaSymbol.Parameters[i].ExplicitDefaultConstantValue;
+                    var delegateParamDefaultVal = delegateParameters[i].ExplicitDefaultConstantValue;
```

Comment:
>delegateParamDefaultVal [](http://example.com/codeflow?start=24&length=23)

Consider getting the delegate parameter default value only after checking `lambdaParamDefaultVal is not null`.